src/back/spv/writer.rs

```diff
@@ -1334,7 +1381,35 @@ impl Writer {
                             ));
                             id
                         }
-                        //TODO: support `crate::TypeInner::Array { .. }` ?
+                        crate::TypeInner::Array {
+                            base: array_base, ..
+                        } => {
+                            let id = self.id_gen.next();
+                            let var_id = match ir_function.expressions[base] {
+                                crate::Expression::Constant(..) => self.cached[base],
+                                crate::Expression::Compose { .. } => self.cached[base],
```

Comment:
Unfortunately, this is incomplete: there other ways to get an array-typed expression:
  - a `Expression::FunctionArgument` that happens to be an array
  - a `Expression::Load` on a pointer to an array
  - an `Expression::Access*` on a structure (or other composite) that contains an array

There needs to be a more general approach here...

Here is a rough idea. Let's have a generic mechanism here for promoting expressions to variables. If an expression is promoted, then the `Emit` of it becomes an `OpStore`, and any use of it becomes an `OpLoad`, other than the `Access`/`AccessIndex` uses, which do `OpLoad` *after* issuing `OpAccessChain`.

We could build a `FastHashMap<Handle<crate::Expression>, Word>` for storing the variable ids, similar to what the PR does.